yt_dlp/extractor/nytimes.py

```diff
@@ -59,204 +51,546 @@ def get_file_size(file_size):
+                    'formats': formats,
+                    'subtitles': subtitles,
+                    'thumbnails': thumbnails})
+
+            return self.playlist_result(entries, page_id, art_title, art_description)
+
+        block_json = blocks_json.pop()
+        media_id = traverse_obj(block_json, ('sourceId'), get_all=False)
+        publication_date = traverse_obj(
+            block_json, ('firstPublished')) or traverse_obj(article_json, ('firstPublished'))
+        timestamp = parse_iso8601(publication_date) if publication_date else None
+        duration = int_or_none(traverse_obj(
+            block_json, ('duration'), get_all=False), 1000) or traverse_obj(block_json, ('length'), get_all=False)
+        series = traverse_obj(block_json, ('podcastSeries'), get_all=False)
+        uploader = traverse_obj(block_json, ('bylines', ..., 'renderedRepresentation'), get_all=False)
+
+        media_content = traverse_obj(block_json, ('renditions', ...))
+        formats, subtitles = self._extract_media_from_json(media_id, media_content)
+
+        # audio articles won't have formats
+        url = traverse_obj(block_json, ('fileUrl'), get_all=False)
+        if not formats and url:
+            formats.append({'url': url, 'ext': determine_ext(url)})
 
-        episode_title = track['title']
-        video_url = track['source']
+        thumbnails = []
+        for image in traverse_obj(
+            block_json, ('promotionalMedia', 'crops', ..., 'renditions', ...), get_all=True) or traverse_obj(
+                article_json, ('promotionalMedia', 'assetCrops', ..., 'renditions', ...)):
+            image_url = image.get('url')
+            if not image_url:
+                continue
+            thumbnails.append({
+                'url': image_url,
+                'width': int_or_none(image.get('width')),
+                'height': int_or_none(image.get('height')),
+            })
 
-        description = track.get('description') or self._html_search_meta(
-            ['og:description', 'twitter:description'], webpage)
+        return {
+            'id': media_id,
+            'title': art_title,
+            'description': art_description,
+            'timestamp': timestamp,
+            'upload_date': unified_strdate(publication_date),
+            'uploader': uploader,
+            'creator': creator,
+            'duration': duration,
+            'formats': formats,
+            'subtitles': subtitles,
+            'thumbnails': thumbnails,
+            'series': series
+        }
 
-        podcast_title = audio_data.get('podcast', {}).get('title')
-        title = ('%s: %s' % (podcast_title, episode_title)
-                 if podcast_title else episode_title)
 
-        episode = audio_data.get('podcast', {}).get('episode') or ''
-        episode_number = int_or_none(self._search_regex(
-            r'[Ee]pisode\s+(\d+)', episode, 'episode number', default=None))
+class NYTimesCookingReceipesIE(InfoExtractor):
```

Comment:
IMO keep this named `NYTimesCookingIE`
you can add `IE_NAME = NYTimesCookingRecipes` if you want (note the spelling of "Recipes")